### ISSUE TITLE
range(int) flagged as error

### ISSUE THREAD

**Issue by tzachar** [microsoft/pyright#3163]
**Describe the bug**
Started getting the following error: `Argument of type "Literal[10]" cannot be assigned to parameter "__stop" of type "SupportsIndex" in function "__init__"    "Literal[10]"`

Error code is: `reportGeneralTypeIssues`

**To Reproduce**
The following simple code:
```python
for i in range(10):
   pass
```

**Expected behavior**
No error generated

**VS Code extension or command-line**
running in command line, version is 1.1.228

This only started lately, after around versoin 1.1.200 or something. On 1.1.117 I dont get this error.

**Comment by erictraut** [comment]
I'm not able to repro the problem as reported. Are you using a custom version of typeshed stdlib stubs? Please try to repro with a new project with default settings.

**Comment by tzachar** [comment]
running with no configuration and no typeshed:
```
❯ nodejs ~/.local/share/nvim/lsp_servers/python/node_modules/pyright/index.js /tmp/1.py --verbose
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/typings is not a valid directory.
Assuming Python platform Linux
Search paths for /tmp
  /home/tzachar/.local/share/nvim/lsp_servers/python/node_modules/pyright/dist/typeshed-fallback/stdlib
  /tmp
  /tmp/typings
  /home/tzachar/.local/share/nvim/lsp_servers/python/node_modules/pyright/dist/typeshed-fallback/stubs/...
  /tmp
  /home/tzachar/python
  /home/tzachar/work/vault/code/vault_py
  /home/tzachar/work/vault/code/vault_py/apis
  /home/tzachar/work/vault/code
  /home/tzachar/.pyenv/versions/3.8.7/lib/python3.8
  /home/tzachar/.pyenv/versions/3.8.7/lib/python3.8/lib-dynload
  /home/tzachar/.local/lib/python3.8/site-packages
  /home/tzachar/.pyenv/versions/3.8.7/lib/python3.8/site-packages
Searching for source files
Found 1 source file
/tmp/1.py
  /tmp/1.py:1:16 - error: Argument of type "Literal[10]" cannot be assigned to parameter "__stop" of type "SupportsIndex" in function "__init__"
    "Literal[10]" is incompatible with "SupportsIndex" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
Completed in 0.726sec
```

**Comment by erictraut** [comment]
I'm still unable to repro the problem.

I suspect it has something to do with a source or stub file that's located in one of your search paths. Please try to repro with a clean virtual environment.

**Comment by tzachar** [comment]
Running in a clean virtualenv solves the problem. 
How should I go about finding the problematic source / stub?

**Comment by erictraut** [comment]
The most likely source is an old version of types-typing_extensions. See https://github.com/python/typeshed/issues/7197 for details.

**Comment by tzachar** [comment]
Thank you.
This solved the issue
